### ISSUE TITLE
Translate filter incorrectly assumes this===scope

### ISSUE THREAD

**Issue by provegard** [angular-translate/angular-translate#1607]
### Subject of the issue
When using the `translate` filter with a string argument, the filter uses `$parse` to parse the string, then evaluates the expression against `this`, assuming `this` is the current scope. But that is an incorrect assumption since Angular 1.3, see this commit:

https://github.com/angular/angular.js/commit/8863b9d04c722b278fa93c5d66ad1e578ad6eb1f

### Your environment
* version of angular-translate: 2.9.0
* version of angular: 1.5.0
* which browser and its version: Chrome 53

### Steps to reproduce
Example: https://plnkr.co/edit/vBoO2V?p=preview

(The example show the non-working string-parse case and contrasts with the working object case.)

### Expected behaviour
Since it's not possible to evaluate the parsed string against the scope, the filter should probably throw an exception.

### Actual behaviour
The parsed string is evaluated against `this`, which is `undefined`, which means that scope variables are not resolved.

**Comment by knalli** [comment]
I get your point, but: In which case an error should be thrown?

**Comment by provegard** [comment]
Here: 

https://github.com/angular-translate/angular-translate/blob/master/src/filter/translate.js#L61

...since that line of code doesn't work properly in all cases given the parse context will be null/undefined.

**Comment by provegard** [comment]
But sure, it works as long as the parsed object doesn't depend on the scope. Perhaps that's a valid use case?

**Comment by knalli** [comment]
Yeah, I got your point! :) But again: Which case should throw an error? If `this` is `undefined` and we throw an error on `undefined`, well, the filter will stop working effectively.

References in the AngularJS changelog

* [filter: this is undefined](https://github.com/angular/angular.js/commit/8863b9d04c722b278fa93c5d66ad1e578ad6eb1f)
* [filter: no context anymore](https://github.com/angular/angular.js/commit/8863b9d04c722b278fa93c5d66ad1e578ad6eb1f)

I don't see we should do anything here..

**Comment by knalli** [comment]
Should we switch the lined in mention to something like `$parse(interpolateParams)({});`?

**Comment by provegard** [comment]
That has no effect as far as I know, the end result will be the same. But sure, it's clearer than using `this`.

Given ES6 Proxy support, we could catch all property access on the context, and throw a descriptive error in that particular case, but it would only help in modern browsers, so I guess it's not worth the hassle.

Perhaps there's nothing to do about this. But a big note in the documentation would be good!

**Comment by knalli** [comment]
Well, but the `this===undefined` thing is an AngularJS decision. Anyone who is working with filters will be faced with this.

I've no problem replacing it with `{}` and adding notes into the docs. Do you like to take action making a PR?

**Comment by provegard** [comment]
Sure, I'll fix!
